spock-core/src/main/java/org/spockframework/mock/EmptyOrDummyResponse.java

```diff
@@ -35,8 +36,16 @@
 @ThreadSafe
 public class EmptyOrDummyResponse implements IDefaultResponse {
   public static final EmptyOrDummyResponse INSTANCE = new EmptyOrDummyResponse();
+  private final List<IDefaultValueProviderExtension> defaultValueProviders;
 
-  private EmptyOrDummyResponse() {}
+  private EmptyOrDummyResponse() {
+    ServiceLoader<IDefaultValueProviderExtension> serviceLoader = ServiceLoader.load(IDefaultValueProviderExtension.class);
+    List<IDefaultValueProviderExtension> providers = new ArrayList<>();
+    for (IDefaultValueProviderExtension provider : serviceLoader) {
+      providers.add(provider);
```

Comment:
I guess we should add some kind of sorting here, otherwise the execution can be flaky, if two implementors of `IDefaultValueProviderExtension` handle the same types.

How about at least we sort by impl classname.